R/data-combined.R

```diff
@@ -0,0 +1,867 @@
+        # lengths of alternate names and objects should be same
+        names <- validateVectorArgs(names, length(names(dataSets)), type = "character")
+
+        # if any of the alternate names `NULL`, use original names
+        names <- ifelse(is.na(names), names(dataSets), names)
+      }
+
+      # Update private fields for the new setter call
+      #
+      # - extract dataframe and append it to the combined dataframe
+      # - extract original XY data into separate columns
+      # - extract group map
+      # - extract dataset names
+
+      # styler: off
+      private$.dataCombined <- private$.updateDF(private$.dataCombined, private$.dataSetToDF(dataSets, names, groups))
+      private$.dataCombined <- private$.extractXYData(private$.dataCombined)
+      private$.groupMap     <- private$.extractGroupMap(private$.dataCombined)
+      private$.names        <- private$.extractNames(private$.dataCombined)
+      # styler: on
+
+      # set up data transformations
+      self$setDataTransformations(names)
+
+      # for method chaining
+      invisible(self)
+    },
+
+    #' @description
+    #'
+    #' Add simulated data using instance of `SimulationResults` class.
+    #'
+    #' @return `DataCombined` object containing simulated data.
+
+    addSimulationResults = function(simulationResults,
+                                    quantitiesOrPaths = NULL,
+                                    population = NULL,
+                                    individualIds = NULL,
+                                    names = NULL,
+                                    groups = NULL) {
+      # list or a vector of `SimulationResults` class instances is not allowed
+      if (is.vector(simulationResults)) {
+        stop(messages$errorOnlyOneSupported())
+      }
+
+      # validate the object type (`NULL` elements are not accepted)
+      validateIsOfType(simulationResults, SimulationResults, nullAllowed = FALSE)
+
+      # validate argument type and length
+      groups <- validateVectorArgs(groups, length(quantitiesOrPaths %||% simulationResults$allQuantityPaths), type = "character")
+      names <- validateVectorArgs(names, length(quantitiesOrPaths %||% simulationResults$allQuantityPaths), type = "character")
+
+      # Update private fields for the new setter call
+      #
+      # - extract dataframe and append it to the combined dataframe
+      # - extract original XY data into separate columns
+      # - extract group map
+      # - extract dataset names
+
+      # styler: off
+      private$.dataCombined <- private$.updateDF(private$.dataCombined,
+        private$.simResultsToDF(
+          simulationResults = simulationResults,
+          quantitiesOrPaths = quantitiesOrPaths,
+          population        = population,
+          individualIds     = individualIds,
+          names             = names,
+          groups            = groups
+        )
+      )
+      private$.dataCombined <- private$.extractXYData(private$.dataCombined)
```

Comment:
I have seen the exact same code before. Create a private method that will always set those fields (groupmap, names, extractXY data) at the same time.